backend/rest/rest.go

```diff
@@ -36,49 +36,61 @@ func New(token string, resultsPerPage int, storageProvider persistence.Storage)
 }
 
 // Search searches code using the github api
-func (c *client) Search(query string, lang string) []string {
+func (c *client) Search(query string, lang string) map[int64]*backends.Result {
 
 	opt := &github.SearchOptions{
 		ListOptions: github.ListOptions{PerPage: c.resultsPerPage},
 	}
 
-	var repos []string
 	results, _, err := c.Client.Search.Code(c.ctx, query, opt)
+	repos := make(map[int64]*backends.Result)
+
 	if err != nil {
 		log.Fatal("Failed to fetch results", err)
 	}
 
-	for _, result := range results.CodeResults {
-		repos = append(repos, result.GetRepository().GetHTMLURL())
-		repoID := result.GetRepository().GetID()
-		fmt.Printf("Fetching value for: %v", repoID)
+	for _, codeResult := range results.CodeResults {
+		repoID := codeResult.GetRepository().GetID()
+		fmt.Printf("Fetching value for: %v\n", repoID)
 		value, err := storage.Get(repoID)
 		fmt.Printf("Found value: %s for repoID: %v\n", value, repoID)
 
-		if err != nil || value == "" {
-			log.Printf("Value not found in DB, requesting amount of stars...\n")
+		if err != nil || value == nil {
 			repo, _, err := c.Client.Repositories.GetByID(c.ctx, repoID)
```

Comment:
Don't you want to log the error before overriding it?

Reply:
it's logged in db.go, but yeah, I should